### PR TITLE
feat(YNOproject): add presence

### PR DESCRIPTION
**Pull request by joaovitorbf**
## Description 
Add a presence for ynoproject.net that shows which game you are playing and what's your current location in the game.
This pull request now also covers all available pages and modals for the website, plus a "chatting" status with player count.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
image
image

</details>

### CODE REVIEW

File: websites/Y/YNOproject/presence.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+				break;
+			case "UI Theme":
+				presenceData.details = `Changing UI theme for ${titleInfo[0]}`;
+				break;
+			case "Log In":
+				presenceData.details = `Logging in to ${titleInfo[0]}`;
+				break;
+			default:
+				presenceData.details = `Dreaming on ${titleInfo[0]}`;
+				break;
```
Comment (EncryptedDev) [inline websites/Y/YNOproject/presence.ts:37]: Is there a possibility of `details` here getting particularly long when being displayed due to the combination of static and dynamic data? If so, use `PresenceData.state` too

Comment (joaovitorbf) [reply websites/Y/YNOproject/presence.ts:37]: Just confirmed it happens on Answered Prayers.
Will fix.

Comment (joaovitorbf) [reply websites/Y/YNOproject/presence.ts]: Kept the game title as a details and the location/action as the status.
This should keep everything under the max char limit, except if the ingame location has a really long name.

File: websites/Y/YNOproject/presence.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			case "Badges":
+				presenceData.state = "Viewing badges";
+				break;
+			case "Expeditions":
+				presenceData.state = "Viewing expeditions";
+				break;
+			case "UI Theme":
+				presenceData.state = "Changing UI theme";
+				break;
```
Comment (EncryptedDev) [inline websites/Y/YNOproject/presence.ts:36]: Is this really needed

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/Y/YNOproject/presence.ts (hunk 2 above)
Comment (joaovitorbf) [reply websites/Y/YNOproject/presence.ts:36]: You mean having a state for the UI theme changer?
I think that if I'm doing a state for all the available buttons in the game page, having one for the UI theme button is fine.

**Pull request closed without merging**